sstables/sstable_set.hh

```diff
@@ -28,8 +28,21 @@
 
 namespace sstables {
 
-class sstable_set_impl;
-class incremental_selector_impl;
+class incremental_selector_impl {
+public:
+    virtual ~incremental_selector_impl() {}
+    virtual std::tuple<dht::partition_range, std::vector<shared_sstable>, dht::ring_position_view> select(const dht::ring_position_view&) = 0;
+};
+
+class sstable_set_impl {
+public:
+    virtual ~sstable_set_impl() {}
+    virtual std::unique_ptr<sstable_set_impl> clone() const = 0;
+    virtual std::vector<shared_sstable> select(const dht::partition_range& range) const = 0;
+    virtual void insert(shared_sstable sst) = 0;
+    virtual void erase(shared_sstable sst) = 0;
+    virtual std::unique_ptr<incremental_selector_impl> make_incremental_selector() const = 0;
+};
```

Comment:
Why move sstable_set_impl to the header? I don't mind if there's a good reason, it will rarely change so it won't trigger recompilation. But if it's just an accident, let's keep it in .cc.

Reply:
I moved it because as of right now, we need to see the member specification in both sstable_set.cc and compaction_strategy.cc: in sstable_set.cc we are using sstable_set_impl's methods when selecting or modifying the sstable_set, and in compaction_strategy.cc we're implementing `bag_sstable_set` and `partitioned_sstable_set` and we're using its constructor in `leveled_compaction_strategy::make_sstable_set`. I think I could move the bag- and partitioned- implementations but `make_sstable_set` should probably stay where it is